Taxonomy edit create allow homonyms

Atticus29 opened the pull request:
# Description

This PR addresses #2410 by not disabling the submit button even if the form is flagged as invalid. The warnings still display in red text next to the submit button, and there is a confirmation dialog if the issue is that a taxon with the same name already exists.

This PR also fixes the bug where new taxa created using the taxonomy creation page were not inheriting their parent taxon's kingdomName.

# Pull Request Checklist:

# Pre-Approval

- [x] There is a description section in the pull request that details what the proposed changes do. It can be very brief if need be, but it ought to exist.
- [ ] Hotfixes should be branched off of the `master` branch and PR'd using the **merge** option (not squashed) into the `hotfix` branch.
    - N/A
- [ ] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
    - N/A
- [x] All new text is preferably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages), and the [spreadsheet tracking internationalizations](https://docs.google.com/spreadsheets/d/133fps9w2pUCEjUA6IGCcQotk7dn9KvepMXJ2IWUZsE8/edit?usp=sharing) has been updated either with a new row or with checkmarks to existing rows.
- [x] There are no linter errors
- [ ] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
    - N/A
- [x] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [ ] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
    - N/A
- [ ] Comment which GitHub issue(s), if any does this PR address
    - #2410
- [ ] If this PR makes any changes that would require additional configuration of any Symbiota portals outside of the files tracked in this repository, make sure that those changes are detailed in [this document](https://docs.google.com/document/d/1T7xbXEf2bjjm-PMrlXpUBa69aTMAIROPXVqJqa2ow_I/edit?usp=sharing).
    - N/A

# Post-Approval

- [ ] It is the code author's responsibility to merge their own pull request after it has been approved
- [ ] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [ ] If this PR represents a merge into the `hotfix` branch, remember to use the **merge** option (i.e., no squash).
- [ ] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [ ] If this PR represents a merge from the `hotfix` branch into the `master` branch use the **squash & merge** option
  - [ ] a subsequent PR from `master` into `Development` should be made with the **merge** option (i.e., no squash).
  - [ ] **Immediately** delete the `hotfix` branch and create a new `hotfix` branch
  - [ ] increment the Symbiota version number in the symbase.php file and commit to the `hotfix` branch.
- [ ] If the dev team has agreed that this PR represents the last PR going into the `Development` branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place. Follow the release protocol [here](https://github.com/BioKIC/Symbiota/blob/master/docs/release-protocol.md).
- [ ] Don't forget to delete your feature branch upon merge. Ignore this step as required.

Thanks for contributing and keeping it clean!

Atticus29 committed:
add kingdomName to taxon creation, make the taxonomy creation go through a confirmation if not unique, fix translation bug, fix bug where taxonomy editor form was not getting the correct element in the POST array, fix bug where kingdomName is falsy, add try catch to taxon insertion, add error handling if taxon duplicate is present in taxon editor as well, disable submit button when parent taxon is missing, move uniqueness check to the end since it returns true now, fix submit button being enabled upon page load, update js versions for referenced scripts

Atticus29 commented on js/symb/taxa.sharedTaxonomyCRUD.js:
```diff
@@ -83,14 +83,16 @@ async function handleFieldChange(
   submitButton.textContent = translations.BUTTON_CHECKING;
   const isOk = await verifyLoadForm(form, silent, originalForm);
-  if (!isOk) {
+  const shouldDisableButton = false;
```
I just hard coded this here for now. We can always make this configurable in symbini, but I figured that was not wanted at this point, given that portal-specific database configurations make this impossible even upon click of the submission button.

themerekat reviewed (changes requested):
Taxon adding/editing fix worked, but there is no error handling if the addition fails due to indexes  (foreign keys?) not allowing duplicate values. Discussed with @Atticus29 that he will add some error handling so you don't get the dreaded "this page isn't working" error

Atticus29 commented:
> Taxon adding/editing fix worked, but there is no error handling if the addition fails due to indexes (foreign keys?) not allowing duplicate values. Discussed with @Atticus29 that he will add some error handling so you don't get the dreaded "this page isn't working" error

Fixed for taxonomy creation in 4b4a30097. Fixed for taxonomy edit in f8de821ca.

themerekat reviewed (approved):
It works!

themerekat commented:
@egbot , should this be merged into the latest hotfix?

Atticus29 commented:
> * Parent still needs to be enforced as required. If parent is null, it reports that it is needed, but it doesn't block form submission. Submitting enters name into taxa table, but entry into taxstatus table fails. Easiest fix is just html5 tag the field as required.

Fixed in c7156976b.

Atticus29 commented:
> * In parent field, user needs to be limited to the names in the list. If someone enters a name that is not in the autocomplete list, it needs to report that only names in the list are allowed and form submission blocked until they select a name in the list. Also, if user selects a valid name in the parent list (e.g. Pinus), and then enters a name that is not valid (e.g. Pinaceeae), it remembers the Pinus tid. However, enforcing entry to autocomplete values should resolve this.

Actually, it looks like this got fixed in c7156976b as well!

Atticus29 commented:
> * Unitname1 (e.g. genus) and rankid are allow required strings that should block submission. Submitting without rankID produces an error, but submitting without a value in the genus field creates a new record with sciname and unitname1 entered as a empty string. In both cases, an alert popup appears stating that the name is already in the database, which could probably be removed all together. Recommend simply adding "required" for these input elements, but allowing the browser to enforce this.

This should be fixed in 6d806d75b.

Atticus29 commented:
> When taxonomyloader.php form opens, the Submit New Name button is active. When I click the button, this popup displays. ![error_msg](https://private-user-images.githubusercontent.com/1273407/437221156-51672d69-df47-4900-9e95-494dc2dac98d.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU1MzIyNjEsIm5iZiI6MTc0NTUzMTk2MSwicGF0aCI6Ii8xMjczNDA3LzQzNzIyMTE1Ni01MTY3MmQ2OS1kZjQ3LTQ5MDAtOWU5NS00OTRkYzJkYWM5OGQuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MjRUMjE1OTIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDk5MWM4MjFkZjU0ZWNhMDNkNjVlMTJiYjM1ZTlmMzljZGVhNWEyYjA2OTg1MzZlNDUwOTJjNzhmYzViZWY3MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Yr1bGZFwAEhcyTF3zsYPYo61IfIosthGy0Kly0b4Qzs) If I then click OK, it submit form, inserts a new record into taxa table, but not in taxstatus, and then displays this: ![image](https://private-user-images.githubusercontent.com/1273407/437221497-ced898b3-c51e-4685-be52-3e8cff8817e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU1MzIyNjEsIm5iZiI6MTc0NTUzMTk2MSwicGF0aCI6Ii8xMjczNDA3LzQzNzIyMTQ5Ny1jZWQ4OThiMy1jNTFlLTQ2ODUtYmU1Mi0zZThjZmY4ODE3ZTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MjRUMjE1OTIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWYzY2E1ZWFmZjlkNTMxNTA3OTk5MzE5N2Y1ZGM2NTBjODRlN2E0MDFjMzFiYjMyZDA0NGU5M2M1NDgyNWNhYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.tQmKM6KvHL25JCGi7Vnz0cEMU3RwZ_NFyfCwsVyBR-k)

Fixed in 77d6c1c8a, I believe.

Atticus29 commented:
@themerekat @egbot closing this in favor of https://github.com/Symbiota/Symbiota/pull/2440, which PRs into the 3.2.3 Hotfix branch instead. Any objections?
